embedfs/embed_test.go

```diff
@@ -338,10 +325,188 @@ func TestJoin(t *testing.T) {
 
 	for _, tc := range tests {
 		t.Run(tc.name, func(t *testing.T) {
-			fs := New(&empty)
+			fs := New(testdata.GetTestData())
 
 			got := fs.Join(tc.path...)
 			assert.Equal(t, tc.want, got)
 		})
 	}
 }
+
+// Additional comprehensive tests using rich test data
+
+func TestEmbedfs_ComprehensiveOpen(t *testing.T) {
+	t.Parallel()
+	
+	fs := New(testdata.GetTestData())
+	
+	// Test opening existing embedded file with content
+	f, err := fs.Open("/testdata/file1.txt")
+	require.NoError(t, err)
+	assert.Equal(t, "testdata/file1.txt", f.Name())
+	require.NoError(t, f.Close())
+}
+
+func TestEmbedfs_ComprehensiveRead(t *testing.T) {
+	t.Parallel()
+	
+	fs := New(testdata.GetTestData())
+	
+	f, err := fs.Open("/testdata/file1.txt")
+	require.NoError(t, err)
+	defer f.Close()
+	
+	// Read the actual content
+	buf := make([]byte, 100)
+	n, err := f.Read(buf)
+	require.NoError(t, err)
+	assert.Equal(t, "Hello from embedfs!", string(buf[:n]))
+}
+
+func TestEmbedfs_NestedFileOperations(t *testing.T) {
+	t.Parallel()
+	
+	fs := New(testdata.GetTestData())
+	
+	// Test nested file read
+	f, err := fs.Open("/testdata/subdir/nested.txt")
+	require.NoError(t, err)
+	defer f.Close()
+	
+	buf := make([]byte, 100)
+	n, err := f.Read(buf)
+	require.NoError(t, err)
+	assert.Equal(t, "Nested file content", string(buf[:n]))
+}
+
+func TestEmbedfs_PathNormalization(t *testing.T) {
+	t.Parallel()
+	
+	fs := New(testdata.GetTestData())
+	
+	// Test that our path normalization works across all methods
+	tests := []struct {
+		name string
+		path string
+	}{
+		{"root", "/"},
+		{"top-level", "/testdata"},
+		{"nested", "/testdata/subdir"},
+		{"deep file", "/testdata/subdir/nested.txt"},
+	}
+	
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// All these should work with our path normalization
+			_, err := fs.Stat(tt.path)
+			if tt.name == "deep file" {
+				require.NoError(t, err, "file should exist")
+			} else {
+				require.NoError(t, err, "directory should exist")
+			}
+		})
+	}
+}
```

Comment:
Replace this with a new test cases within `TestStat`, as the coverage is largely the same.